ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/cloudinit/DefaultCcmV2JumpgateParameterSupplier.java

```diff
@@ -21,10 +22,16 @@
     public CcmV2JumpgateParameters getCcmV2JumpgateParameters(@Nonnull String accountId, @Nonnull Optional<String> environmentCrnOpt,
         @Nonnull String clusterGatewayDomain, @Nonnull String agentKeyId) {
 
-        LOGGER.debug("Returning CCMV2 Jumpgate parameters");
-        CcmV2Parameters ccmV2Parameters = getCcmV2Parameters(accountId, environmentCrnOpt, clusterGatewayDomain, agentKeyId);
-        return new DefaultCcmV2Parameters(ccmV2Parameters.getInvertingProxyHost(), ccmV2Parameters.getInvertingProxyCertificate(),
-                ccmV2Parameters.getAgentCrn(), agentKeyId, ccmV2Parameters.getAgentEncipheredPrivateKey(), ccmV2Parameters.getAgentCertificate());
-    }
+        InvertingProxyAndAgent invertingProxyAndAgent = getInvertingProxyAndAgent(accountId, environmentCrnOpt, clusterGatewayDomain, agentKeyId);
+        InvertingProxy invertingProxy = invertingProxyAndAgent.getInvertingProxy();
+        InvertingProxyAgent invertingProxyAgent = invertingProxyAndAgent.getInvertingProxyAgent();
+
+        LOGGER.debug("CcmV2JumpgateConfig successfully retrieved InvertingProxyHost: '{}', InvertingProxyStatus: '{}', InvertingProxyAgentCrn: '{}', " +
+                        "EnvironmentCrnOpt: '{}'", invertingProxy.getHostname(), invertingProxy.getStatus(),
+                invertingProxyAgent.getAgentCrn(), Optional.of(invertingProxyAgent.getEnvironmentCrn()));
```

Comment:
What is the purpose of wrapping the `environmentCrn` in an `Optional`? If the CRN is never `null`, it could be directly printed, otherwise `Optional.of()` will fail.

Reply:
I did not touch this logic and I am not wanting to, in this PR.